wgpu-core/src/device/trace.rs

```diff
@@ -40,9 +40,18 @@ impl ProgrammableStageDescriptor {
     pub fn new(desc: &crate::pipeline::ProgrammableStageDescriptor) -> Self {
         ProgrammableStageDescriptor {
             module: desc.module,
-            entry_point: unsafe { std::ffi::CStr::from_ptr(desc.entry_point) }
-                .to_string_lossy()
-                .to_string(),
+            entry_point: desc.entry_point.to_string(),
+        }
+    }
+}
+
+impl<'a> From<&'a ProgrammableStageDescriptor>
```

Comment:
let's do these great hacks:)
instead, just have a method in `ProgrammableStageDescriptor` called `to_core()` or something